Extension web - Protect vault with user password & Single extension popup

Chralu opened the pull request:
Fixes #940, #931 


AEWallet extension runs only one Popup for all DApps. 
Hive vault is secured by user password.

# Description

On chrome with AEWallet extension installed.
When using simultaneously multiple Web DApps, each DApp will use a different AEWallet popup.

This PR ensures only one AEWallet popup is opened. The popup will handle all Web Dapps requests.



Related to https://github.com/archethic-foundation/libjs/pull/193

## Material used:

Please delete options that are not relevant.
- Browser

## How Has This Been Tested?

Setup a Chrome browser with AEWallet extension installed.
Run two Web DApps.

Each DApp should successfully perform RPC requests.

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

Chralu committed:
🚧 Improve AEWalletBrowserExtension ergonomy

DApps connect directly to AEWallet extension popup.
AEWallet extension reuses existing popup.

redDwarf03 committed:
🔧 web generation

redDwarf03 committed:
💄 Show wallet when user should validate RPC request

redDwarf03 committed:
⬆️ Update dependencies

redDwarf03 committed:
🔧 Update conf to avoid CSP error

redDwarf03 committed:
⬆️ Update dependencies

redDwarf03 committed:
🔧 Just password in web mode

redDwarf03 committed:
💄 Adapt background for web platform

redDwarf03 committed:
🚧 Protect vault with password

redDwarf03 committed:
🔧 Conf flutter cache management

redDwarf03 committed:
🚧 Secure hive in web mode

Chralu committed:
🚧 Add password storage encryption.

Chralu committed:
🚧 Vault lock/unlock requesting user password/pin on web.

Today, pin/password is stored in the Vault. But Vault unlock is required to check password/pin validity.

provide two password/pin verification methods.

On web, try to unlock Vault. If succeeds, then provided password/pin is valid

On other platforms, use current method (compare entered password with the one stored in Vault).

redDwarf03 committed:
🔧 Remove unused permission

redDwarf03 committed:
🔖 Send 2.3.8.526 to chrome store

Chralu committed:
🚧 Fix wallet import and restoration.

Chralu committed:
🚧 Remove non-web authent data from Vault.

Chralu committed:
🚧 Store password in secure_storage without supplementary encryption.

Chralu committed:
🐛 Autolock repaired.

Chralu committed:
🚨 [ChromeExt] Fix lint issues.

redDwarf03 commented:
Interactions with DApps tested with aeBridge and aeSwap in testnet env.

Problems encountered:
- [ ] : Logout : after password filled, a black screen is displayed instead of welcome screen
- [ ] : Pool list doesn't appear in aeSwap when aeWallet is in intro process. And when the intro is finished (wallet imported for example), the list appears automatically in aeSwap
- [ ] : UI (Bonus) : is it possible to display wallet just under chrome extension button instead of on the left side of the browser

No issue to bridge, swap.
No issue after password update

Chralu committed:
💄 Set password form can be validated with "return" key.

Chralu committed:
📝 Document local datasources
🗃️ Migrate all Authentication data to a dedicated HiveSecuredBox.

Chralu committed:
🐛 Fix async missing await.

Chralu committed:
🐛  Hive box deletion now compatible with Web.
🐛 Hive box can be deleted event if they are closed.

This is a workaround to https://github.com/isar/hive/issues/344.

Chralu committed:
⚰️ Remove dead code.

Chralu committed:
💄 Display a waiting screen during wallet deletion.

Chralu committed:
♻️ Split AppDB into HiveDatasource classes.

Chralu committed:
♻️ Use logout method to remove code duplication

Chralu committed:
🐛 Fix Hive box deletion type error.

Chralu committed:
💄 LoggingOut page displays during 2 seconds.

Chralu committed:
📝 Update data storage infos.

Chralu committed:
♻️ Renamed WalletProvider file to SessionProvider.

Chralu committed:
Improve Hive.deleteBox workaround to actually delete boxes themselves.

Until now we were only emptying boxes.

Chralu committed:
🐛 Fix Vault unlock issues on non web platforms.

Chralu committed:
🚨 Fix browser extension lints.

Chralu committed:
🐛 Wallet Extension `ensureExtensionPopupOpened` does not wait for RPC to be listening.

Chralu committed:
💄 Extension popup opens at the center of the browser window.

Chralu committed:
🐛 Ensures extension popup is a singleton.

Chralu committed:
🐛 FIx BrowserExtensionRPC ports release.

Chralu committed:
🐛 FIx BrowserExtensionRPC opening status detection.

redDwarf03 committed:
💄 Add icon

redDwarf03 committed:
👷 build web

Chralu committed:
Import Archethic Wallet Client in web  browser extension implementation.

Chralu committed:
🚨 Fix a fez Background.ts lints.
